### PR TITLE
feat(reindex): consume index ranges

### PR DESCRIPTION
**Pull request by psmagin**
### Purpose
Consume index ranges and index records from db to search engine.

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [ ] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [ ] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
_List any Jira issues related to this pull request._
[MSEARCH-800](https://folio-org.atlassian.net/browse/MSEARCH-800)

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -145,6 +145,10 @@ folio:
         concurrency: ${KAFKA_LINKED_DATA_CONCURRENCY:1}
         topic-pattern: (${folio.environment}\.)(.*\.)linked-data\.(work|authority)
         group-id: ${folio.environment}-mod-search-linked-data-group
+      reindex-range-index:
+        concurrency: ${KAFKA_LINKED_DATA_CONCURRENCY:1}
```
Comment (viacheslavkol) [inline src/main/resources/application.yml:149]:
```suggestion
        concurrency: ${KAFKA_REINDEX_RANGE_INDEX_CONCURRENCY:1}
```

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -109,7 +109,7 @@ folio:
       - name: search.consortium.instance
         numPartitions: ${KAFKA_CONSORTIUM_INSTANCE_TOPIC_PARTITIONS:50}
         replicationFactor: ${KAFKA_CONSORTIUM_INSTANCE_TOPIC_REPLICATION_FACTOR:}
-      - name: search.reindex.range.index
+      - name: search.reindex.range-index
         numPartitions: ${KAFKA_REINDEX_RANGE_INDEX_TOPIC_PARTITIONS:16}
         replicationFactor: ${KAFKA_CONSORTIUM_INSTANCE_TOPIC_REPLICATION_FACTOR:}
```
Comment (viacheslavkol) [inline src/main/resources/application.yml:114]:
```suggestion
        replicationFactor: ${KAFKA_REINDEX_RANGE_INDEX_TOPIC_REPLICATION_FACTOR:}
```

File: src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml [hunk 3]
```diff
@@ -210,14 +210,17 @@
     <comment>Create upload_ranges table</comment>
 
     <createTable tableName="upload_range">
-      <column name="entity_type" type="VARCHAR(30)">
+      <column name="id" type="UUID">
         <constraints nullable="false" primaryKey="true" primaryKeyName="pk_upload_ranges"/>
```
Comment (viacheslavkol) [inline src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml:214]:
```suggestion
        <constraints nullable="false" primaryKey="true" primaryKeyName="pk_upload_range"/>
```

**Review by viacheslavkol — CHANGES_REQUESTED** [review]

**Comment by viacheslavkol** [comment]
Don't we need to mention it in NEWS?

File: src/main/java/org/folio/search/service/reindex/ReindexOrchestrationService.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  private final ReindexRangeIndexService rangeIndexService;
+  private final PrimaryResourceRepository elasticRepository;
+  private final MultiTenantSearchDocumentConverter documentConverter;
+
+  public boolean process(ReindexRangeIndexEvent event) {
+    var resourceEvents = rangeIndexService.fetchRecordRange(event);
+    var documents = documentConverter.convert(resourceEvents).values().stream().flatMap(Collection::stream).toList();
+    var folioIndexOperationResponse = elasticRepository.indexResources(documents);
+    if (folioIndexOperationResponse.getStatus() == FolioIndexOperationResponse.StatusEnum.ERROR) {
+      throw new ReindexException(folioIndexOperationResponse.getErrorMessage());
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/service/reindex/ReindexOrchestrationService.java:25]:
```suggestion
     // TODO MSEARCH-797 - update status as failed indicating upload has failed
      throw new ReindexException(folioIndexOperationResponse.getErrorMessage());
```

File: src/main/java/org/folio/search/service/reindex/jdbc/ClassificationJdbcRepository.java [hunk 5]
```diff
@@ -1,24 +1,44 @@
...
 
   @Override
   public ReindexEntityType entityType() {
     return ReindexEntityType.CLASSIFICATION;
   }
 
+  @Override
+  protected RowMapper<Map<String, Object>> rowToMapMapper() {
+    return (rs, rowNum) -> {
+      Map<String, Object> subject = new HashMap<>();
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/service/reindex/jdbc/ClassificationJdbcRepository.java:30]:
```suggestion
      Map<String, Object> classification = new HashMap<>();
```

File: src/main/java/org/folio/search/service/reindex/jdbc/ContributorJdbcRepository.java [hunk 6]
```diff
@@ -1,24 +1,45 @@
...
 
   @Override
   public ReindexEntityType entityType() {
     return ReindexEntityType.CONTRIBUTOR;
   }
 
+  @Override
+  protected RowMapper<Map<String, Object>> rowToMapMapper() {
+    return (rs, rowNum) -> {
+      Map<String, Object> subject = new HashMap<>();
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/service/reindex/jdbc/ContributorJdbcRepository.java:30]:
```suggestion
      Map<String, Object> contributor = new HashMap<>();
```

File: src/main/java/org/folio/search/service/reindex/jdbc/ReindexJdbcRepository.java [hunk 7]
```diff
@@ -73,11 +83,28 @@ public Integer countEntities() {
...
+    return jdbcTemplate.query(sql, rowToMapMapper(), limit, offset);
+  }
+
+  public void setIndexRangeFinishDate(UUID id, Timestamp timestamp) {
+    var sql = UPDATE_FINISHED_AT_UPLOAD_RANGE_SQL.formatted(getFullTableName(context, entityTable()));
+    jdbcTemplate.update(sql, timestamp, id);
+  }
+
+  protected String getFetchBySql() {
+    return "SELECT * from %s LIMIT ? OFFSET ?".formatted(getFullTableName(context, entityTable()));
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/service/reindex/jdbc/ReindexJdbcRepository.java:97]: maybe ```;``` is needed at the end of select query

**Review by viacheslavkol — APPROVED** [review]

**Review by mukhiddin-yusuf — APPROVED** [review]

**Pull request merged**